### PR TITLE
docs: Update minimum TIFF version to 4.0.3

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Companion to https://github.com/ome/ome-files-cpp/pull/29 which documents the change.

Staged here: http://www.openmicroscopy.org/site/support/ome-files-cpp-staging/ome-cmake-superbuild/manual/html/packages.html

### CODE REVIEW

File: docs/sphinx/maintenance.rst [hunk 1]
```diff
@@ -153,12 +153,6 @@ purposes.
...
 dropped.  Examples include:
 
-- dropping support for older versions of libraries, e.g. we have
-  extensive workarounds to support older libtiff releases which can be
-  removed entirely once all the Linux distribution versions we support
-  are 4.0.0 or later, or 4.0.4 or later (there are two sets of
-  workarounds); this will simplify the code, and also allow
-  assumptions to be made regarding the availability of BigTIFF support
 - dropping functionality checks and workarounds for functionality and
   behaviour differences, e.g. missing filesystem functions and
```
Comment (mtbc) [inline docs/sphinx/maintenance.rst]: behavior

File: docs/sphinx/packages/tiff.rst [hunk 2]
```diff
@@ -19,4 +19,4 @@ file format, used for reading and writing TIFF images.
 +------------------+----------------+
 
 Versions 4.0.2 and earlier do not have the :cpp:func:`TIFFField`
-accessor functions.
+accessor functions, and so 4.0.3 or later is required.
```
Comment (mtbc) [inline docs/sphinx/packages/tiff.rst]: maybe lose "and"

**Comment by sbesson** [comment]
@rleigh-codelibre: can you push the 2 minor fixes prior to merging?

**Comment by rleigh-codelibre** [comment]
Done.

**Comment by sbesson** [comment]
Renders as expected. Thanks.

**Pull request merged**